source/docs/software/commandbased/commands-v3/index.rst

```diff
@@ -0,0 +1,115 @@
+# Commands v3 Programming
+
+This section serves as an introduction to and reference for the WPILib Commands v3 framework.
+
+Commands v3 is the next-generation command-based framework for Java that introduces imperative-style command writing using coroutines. Instead of building complex chains of lambda expressions and decorators, you can write command logic as straightforward sequential code with loops, conditionals, and explicit control flow.
+
+**WPILib recommends Commands v3 for Java teams.** Commands v3 is actively being developed with new features including improved telemetry, enhanced trigger functionality, and self-canceling commands.
+
+.. warning::
+   Commands v3 is **Java-only** and does not support C++ or Python.
```

Comment:
Somewhere it might be good to document why it's java only, but probably not here.